packages/graphql/src/plugins/nexusMaxExecutionPlugin.ts

```diff
@@ -0,0 +1,74 @@
+import type { DataContext } from '@packages/data-context'
+import { DirectiveLocation, GraphQLDirective } from 'graphql'
+import { plugin } from 'nexus'
+import type { NexusGenFieldTypes } from '../gen/nxs.gen'
+
+interface MaxExecutionConfig {
+  duration: number
+  triggerOnResult?: keyof NexusGenFieldTypes['Subscription']
+}
+
+export const maxExecutionDirective = new GraphQLDirective({
+  name: 'skipMaxExecutionGuard',
+  description: 'Skips any maxExecution guard defined internally on the field',
+  locations: [DirectiveLocation.QUERY],
+  isRepeatable: false,
+})
+
+// Added here because the field is defined automatically by the merged schema
+const CLOUD_EXECUTION: Record<string, Record<string, MaxExecutionConfig>> = {
+  Query: {
+    cloudViewer: {
+      duration: 1000,
+      triggerOnResult: 'cloudViewerChange',
+    },
+  },
+}
```

Comment:
hmm, shouldn't this go where the query itself is defined?

Reply:
It will typically go on where the field is defined, but this field is defined dynamically because we're stitching in the GraphQL schema from the dashboard automatically.